### DISCUSSION TITLE
Monthly email for inProgress and unread articles

### DISCUSSION THREAD

**Discussion by Roznoshchik** [Roznoshchik/Lurnby#33] [Ideas]
Once a month an email might come out with something along the lines of:

Hey, here are some articles you might want to look at.

Started:
- The power of cats, added on Jan 3, 2999, progress 48%, [ keep reading ]
- Plants and things, added on Sep 17, 2999, progress 8%, [ keep reading ] 
- An epistemology of wigs, added on Feb 1, 2999, progress 1%, [ keep reading ] 
- How to lose friends and be hated by people, added on jan 15, 2999, progress 13%, [ keep reading] 

New:
- new new, added Sep 3, 2999 [ start reading ] 
- the very new, added Mar 12, 2999 [ start reading ] 
- only the newest things here, added Oct 8, 2999 [ start reading ] 

This would make it easy to resurface things you added that you forgot about.

@sharedphysics think this is worth pursuing?

**Comment by sharedphysics** [comment]
I don’t have too strong of a view right now. Perhaps it would be useful to reframe it as:

- What problem/pain/opportunity is this solving for the user?
- What problem/pain/opportunity is this solving for Lurnby as a service/business?
- Why is this a problem/opportunity? (Justification)
- How big of a problem/opportunity is it?
- How would you know when you solved it?

If we start with the problem/opportunity, then we can better suggest changes or provide commentary around the feature proposal :)

**Reply by Roznoshchik** [reply]
Good points. 

## Problem / Opportunity
The problem here is that it becomes increasingly harder to discover things you added to Lurnby once your articles have accumulated past a certain amount. 

<img width="383" alt="image" src="https://user-images.githubusercontent.com/45914541/192047006-14b4c20e-c58d-4478-9a35-a4f06a0de61a.png">

I for instance have over 200 unfinished/unread articles. Lurnby defaults to showing me 15 at a time on the main page, and unless I KNOW what I am looking for, I don't really move past the first or second page of articles when looking for something to read.

## Technical difficulty
Think this fairly simple to get going. 

We can piggyback off of the existing cron job that checks if `recent highlights emails` need to be sent out, by just giving it another function call to check if `article emails` need to be sent out.

Then it's literally just querying and sampling a users existing articles for `unread` or `inProgress` and then scheduling an email to be sent. I think we can reuse much of the `recent highlights` code and simplify it for this purpose.

## Time
I would say that it's about 1 days effort to get this off the ground. Including writing any tests for it.

**Comment by sharedphysics** [comment]
Ok, so a few additional solutions come to mind:

- Most services nowadays offer infinite scroll rather than pagination. The trade off that I remember is that infinite scroll gets people to go further, but pagination encourages them to engage with one of the options earlier. But infinite scroll would make it easy to go further and further for discovery of older stuff.
- You could also build a “surprise me” button. Whereas infinite scroll is reverse chronological, “surprise me” is random. You could also do a surprise full-page assortment. You could also do a surprise full page assortment by tag/topic/source. Reorganizing information is a great way to see it in a new light and rekindle interest or draw new lines between different content.
- But I want to also keep in line with the core of Lurnby. Why do people read in here rather than in other tools such as pocket or rss reader, or something else? It’s because they are reading to learn and to study/review. So the best feature is one that helps with rediscovery and also aligns with helping people get more value. If people are using tags, then I would highlight tag-related pieces based on what they are reading. If you want to be really fancy, you can parse all of the content with an AI reader (say, from Google Cloud Platform) and generate your own tags-2 that you can use to (a) suggest tags to the user or (b) surface content for discoverability.

The problem with an email is that you have ~200 articles unread or partially read already. Are all of those going to make it into the email each week? (That would be a long email). What is your selection criteria? If it is recency or time based, you’re still going to generate an archive of stuff hidden by time.

An email could be good, but I see the benefits of that being engagement based… I.e., “you added X new articles this week but didn’t read them!” But anecdotally, sometimes saving something to read later is just a cognitive trick to allow you to close an otherwise interesting tab without feeling bad that you’re skipping something interesting for now. A lot of people talk about their unread articles in pocket or Instapaper, or their unread pile of New Yorker magazines, their growing list of unread emails, etc. That’s normal behavior. More goes in than gets closed out. So I think the “right” solution for Lurnby is one that creates some unique value by solving the problem. Whereas something like Pocket or Instapaper has no strong opinion on what you do with the content, Lurnby does. Leaning into that is a superpower for creating differentiated features.

**Reply by Roznoshchik** [reply]
Infinite scroll is a good idea regardless. I think that should be done. The pagination was initially introduced as a way speed up load times with the understanding that the majority of the time, someone probably just wants to keep reading. 

But infinite scroll would help with discovery and keep load times the same, as we would just need to fetch a small amount of articles with every load. 

As far as the rest of the ideas go, I think you are spot on in that the spirit of lurnby is to give you more value than other tools. Mainly, I think of the underlying goal as "making it as easy as possible for you to do things that maximize your benefit when it comes to learning".

In terms of LCD/MVP The selection would be limited to around 10 articles, 5 unread and 5 in progress.

But in terms of more interesting stuff I like the idea of showing tagged content - the issue is that not everyone uses tags. 

One of my other plans was to get rid of the concept of "tags" and "topics" and just make it a single thing "tags". But that's a different discussion.

That would maybe make the segmentation of articles a bit easier, if we could also be checking against what the highlights are being tagged with.

**Reply by sharedphysics** [reply]
> "making it as easy as possible for you to do things that maximize your benefit when it comes to learning".

A version of this should be plastered on the ReadMe & marketing homepage as the north star.

I know this is getting a bit off topic here, but that's a really important sentiment. I want to poke at it a bit more... There are three types of learning, right?

- **structured** (where you are working on a very specific topic in an organized fashion, i.e., learning to code or learning biochem 101), 
- **unstructured** (where you're not following any sort of learning plan but you still want to maximize the "learning" value you get from content consumption), and
- **semi-structured** (where you have a topic you're interested in and you're consuming related content around it, but not following a formal plan).

The audience, behaviors, and needs around each of these is pretty different. They overlap to some degree ("learning-equipping" traditional content consumption), but mostly different. If you're learning in a structured way, then you need ways to recreate that     structure. Something like "Tags" (Primary/Secondary Source, Article/Paper) and "Topics" (BioChem, Finance, Digital Advertising) are both useful and coexist. If you're trying to make content consumption learning-enabled, then that structure is confusing and gets in the way. People use features that have a clear purpose and that are in their. If the feature isn't clear as to how you can use it or the value is hidden/disconnected from it's immediate use, then only power users will end up using it if at all.

So this tangent is really about: in the spirit of different kinds of learning goals/needs, who are your users and what are they trying to do? If you have a clear answer to that, then the answer to "is this a valuable feature" or "how should we implement this" or "how do we know if this actually created value?" becomes obvious. It _also_ helps then with marketing and positioning because you know exactly who to go to and what to tell them, as well as who to do research with to better adapt the solution.

Right now, the development roadmap feels a bit like "these are things I want to make based on problems I encounter through my personal and evolving learning journey". That doesn't make for a cohesive vision or a clear value prop. It's not a criticism (I know I do plenty of that myself too), but just an outside perspective trying to rationalize what I'm seeing. 

For example, 

- If you tell me that your target user is a structured learning student of some sort, then I'd say "make tags and topics a mandatory part of the article-saving/review process because they will inevitably benefit form that structure, and then you can also leverage that structure to develop other features  (such as emails that highlight articles you may have missed within a topic you're studying deeply, or a "progress bar" showing how much of a topic you've studied based on articles you've saved vs read), or new way of organizing the homepage of articles). 
- If you say that your target user is recreational, unstructured content consumer who wants to not forget things they read, then I'd say that you can definitely consolidate tags and topics and your email should have a way of marking things as "no longer relevant" when the email surfaces something -- these folks will inevitably save more than they can consume _and their goal is not to consume everything, but to consume the things relevant to them at the point_... so the emails should pull in a random assortment and let folks mark as "no longer interesting" to take it out of their rotation. 

Anyway, I'm writing all of this because we had a conversation a while back that you like strongly opinionated tools and want to make Lurnby in that fashion, but I'm not seeing the strong opinion grounding here ;) I think defining the core user and core problem and core goals will go a long way in helping you, me, and anyone else be more productive and helpful in offering feedback suggestions, feature ideas, and also with user acquisition.

---

Separately from all of that, agree with your other comment. To-dos are:

- [ ] Create feature branch for infinite scroll
- [ ] Create feature branch for emails because I think you want these regardless of what I suggest :)
